### PR TITLE
Get correct needed datum when using child modules

### PR DESCRIPTION
**Pull request by phillipm**
Fix for some issues in http://manage.dimagi.com/default.asp?234442

### CODE REVIEW

File: backend/src/org/commcare/session/CommCareSession.java [hunk 1]
```diff
@@ -125,22 +126,21 @@ public CommCareSession(CommCareSession oldCommCareSession) {
...
+        Vector<Entry> entries = new Vector<>();
         for (Suite s : platform.getInstalledSuites()) {
-            for (Menu m : s.getMenus()) {
-                // We need to see if everything in this menu can be matched
-                if (commandId.equals(m.getId())) {
-                    return getEntriesFromMenu(m, currentSessionData);
+            List<Menu> menusWithId = s.getMenusWithId(commandId);
+            if (menusWithId != null) {
+                for (Menu menu : menusWithId) {
+                    entries.addAll(getEntriesFromMenu(menu, currentSessionData));
```
Comment (amstone326) [inline backend/src/org/commcare/session/CommCareSession.java]: Is this the fix? curious how this resolves Kriti's issue of displaying the wrong menu name. Or is that not what this PR is addressing? (tried to read through the ticket you linked but there were a few different things going on there)

Comment (phillipm) [inline backend/src/org/commcare/session/CommCareSession.java]: This PR resolves a session ordering issue with when a case list is displayed. If there are 2 forms that use different case lists and are associated with a single menu command, then it used to just take the first one, which resulted in the case list being shown before form selection.

Comment (amstone326) [inline backend/src/org/commcare/session/CommCareSession.java]: Gotcha, thanks. Do we have plans to resolve the menu naming issue, just out of curiosity?

Comment (phillipm) [inline backend/src/org/commcare/session/CommCareSession.java]: Not sure, I wanted to get the big ones out of the way, but I'll reasses that issue soon. Might be the case that Kriti can use advanced modules instead.

**Pull request merged**